invenio_users_resources/records/api.py

```diff
@@ -367,6 +397,49 @@ def get_record_by_name(cls, name):
                 return None
             return cls.from_model(role)
 
+    @classmethod
+    def create(cls, data):
+        """Create a new group/role and store it in the database."""
+        try:
+            # Validate before attempting DB write to avoid UOW commit failures
+            _validate_group_data(data)
+            # Create Role
+            role = current_datastore.create_role(**data)
+            return cls.from_model(role)
+        except ValidationError:
+            raise
+        except IntegrityError:
+            # Unique constraint likely violated; re-raise as a field error
+            raise ValidationError(
+                {"name": [_("Role name already used by another group.")]}
+            )
+        except Exception as e:
+            raise ValidationError(f"Unexpected error creating group: {str(e)}")
+
+    def update(self, data):
+        """Update the group/role attributes.
+
+        Update is proxied through direct attribute modification.
+        """
+        role = self.model.model_obj
+        if role is None:
+            raise ValueError("Cannot update group without an underlying model.")
+
+        if "description" in data:
+            role.description = data["description"]
```

Comment:
We might also want to do the dance with the name `role.name = data.get("name", role.name)`

Reply:
I don't see any reason why we wouldn't ☺️